src/pydata_sphinx_theme/assets/scripts/fontawesome.js

```diff
@@ -1,5 +1,5 @@
 // TODO: @trallard might end up moving this to the main JS file
 // Define the custom behavior of the page
-import { documentReady } from "./mixin";
 
+import "@fortawesome/fontawesome-free/js/fontawesome.min.js";
```

Comment:
Why is this import needed? 

Testing locally, if I remove `fontawesome.min.js` (but leave `all.min.js`) everything looks right on the home page. If I remove `all.min.js` (but leave `fontawesom.min.js`) the home page is missing icons. 

Seems like only `all.min.js` needs to be imported here?

Reply:
I followed the docs at https://docs.fontawesome.com/web/setup/host-yourself/svg-js (this is also what we had before).

I think because `fontawesome.js` is the loader, we need that to allow for custom icons to be added through `FontAwesome.library.add()` `all.min.js` should only be the SVG icons.

Reply:
Hmm. My knowledge of Font Awesome is shaky but I think the approach we're taking now is some kind of hybrid of the [package manager](https://docs.fontawesome.com/web/setup/packages) and svg-js approaches. 

There is a [Font Awesome Python/Django docs page](https://docs.fontawesome.com/web/use-with/python-django#alternate-install-using-allcss-or-alljs) that says:

> We have also provided a shortcut all.min.js or all.min.css file that includes all the Free icons as well as the core utility functions and styles. It’s a handy shortcut but isn’t as performant as selecting individual styles.

My understanding is that because we advertise PST as "Font Awesome Free included," we have to:

- Include all three of the free icon sets (regular, solid, and brand) 
- Include both the svg+js and web fonts+css methods of using those icons, which means we have to ship both all.js and all.css down the wire to the end user.

Reply:
> Hmm. My knowledge of Font Awesome is shaky but I think the approach we're taking now is some kind of hybrid of the [package manager](https://docs.fontawesome.com/web/setup/packages) and svg-js approaches.

This is correct. We are doing some things from one and some from the other. This means we are vendoring both the JS and CSS FA variants (which have performance and maintenance overheads).

> Include both the svg+js and web fonts+css methods of using those icons, which means we have to ship both all.js and all.css down the wire to the end user.

That is incorrect. We only need one or the other from the Python docs you linked to:

> We have also provided a shortcut all.min.js **OR** all.min.css file that includes all the Free icons as well as the core utility functions and styles.

The _or_ is the crucial part; regardless of how we vendor FA, the CSS or JS approach provides the same functionalities and end-user experience (syntax and use of the icons are the same functionally).